examples/src/bin/ech-client.rs

```diff
@@ -99,6 +217,29 @@ fn lookup_ech_configs(resolver: &Resolver, domain: &str) -> pki_types::EchConfig
                 }),
             _ => None,
         })
-        .expect("missing expected HTTPS SvcParam EchConfig record")
-        .into()
+        .map(Into::into)
+}
+
+fn read_ech(path: &str) -> pki_types::EchConfigListBytes<'static> {
+    let file = fs::File::open(path).unwrap_or_else(|_| panic!("Cannot open ECH file: {path}"));
+    let mut reader = BufReader::new(file);
+    let mut bytes = Vec::new();
+    reader
+        .read_to_end(&mut bytes)
+        .unwrap_or_else(|_| panic!("Cannot read ECH file: {path}"));
+    bytes.into()
 }
+
+/// A HPKE suite to use for GREASE ECH.
+///
+/// A real implementation should vary this suite across all of the suites that are supported.
+static GREASE_HPKE_SUITE: &dyn Hpke = aws_lc_rs::hpke::DH_KEM_X25519_HKDF_SHA256_AES_128;
+
+/// Randomly generated X25519 public key to use for GREASE ECH.
+///
+/// A real implementation should vary this pubkey, ensuring the public key type matches the KEM
+/// selected in the GREASE [Hpke] instance.
+const GREASE_25519_PUBKEY: &[u8] = &[
```

Comment:
Is it worth extending our Hpke traits to be able to generate fresh meaningless public keys, so we can remove this parameter from `EchGreaseConfig::new()`?

Reply:
That sounds reasonable to me. It might come in handy for https://github.com/rustls/rustls/issues/1980 as well.

Reply:
I'm still chewing on this. It's a little bit tricky, because we're using [`agreement::PrivateKey::from_bytes`](https://docs.rs/aws-lc-rs/latest/aws_lc_rs/agreement/struct.PrivateKey.html#method.from_private_key) to reconstitute a private key from the provided bytes (this is how the test vectors are represented), but the [trait impl](https://docs.rs/aws-lc-rs/latest/aws_lc_rs/agreement/struct.PrivateKey.html#trait-implementations) to go from a generated `PrivateKey` to raw BE bytes is different depending on the type of key: one supports only X25519 and one supports the P curves. I think I can get this working but wanted to push the other easier feedback first.

Reply:
Done. It's a separate commit up-front, and could be lifted into its own PR if desired.